src/download_and_filter.py

```diff
@@ -198,21 +199,69 @@ def apply_filters(
         ]
 
     if not filtered_df.empty and selected_license_use:
-        use_key = "License Use (DataProvenance IgnoreOpenAI)" if openai_license_override else "License Use (DataProvenance)"
         valid_license_use_idx = constants.LICENSE_USE_TYPES.index(selected_license_use)
-        valid_license_uses = constants.LICENSE_USE_TYPES[:valid_license_use_idx+1]
+        valid_license_uses = [x.lower() for x in constants.LICENSE_USE_TYPES[:valid_license_use_idx + 1]]  # ["academic-only", ...]
+
+        # check if openai license override is selected, if so, remove DataProvenance from sources and add DataProvenance IgnoreOpenAI
+        if openai_license_override:
+            # remove "DataProvenance" from selected_license_sources if openai_license_override is selected and add "DataProvenance IgnoreOpenAI" to selected_license_sources
+            if "DataProvenance" in selected_license_sources:
+                selected_license_sources.remove("DataProvenance")
+                selected_license_sources.append("DataProvenance IgnoreOpenAI")
+
+            # Check that DataProvenance is not in selected_license_sources if openai_license_override is selected
+            assert "DataProvenance" not in selected_license_sources, f"DataProvenance should not be in selected_license_sources: {selected_license_sources}"
+
+        # if "DataProvenance" is "unspecified", we want to use the GitHub license information
```

Comment:
We only want to do this if we set that particular flag. 

So by default we use DataProvenance. If we have a flag that says use GitHub license if DP is "unspecified", then we should do that -- but only if there is in fact a GitHub license. If there is no GitHub license, we should leave it at "Unspecified" from DP.

Otherwise this looks great!